cni-plugin/pkg/plugin/plugin.go

```diff
@@ -104,6 +110,29 @@ func testConnection() error {
 	return nil
 }
 
+func isEndpointReady(readyEndpoint string) (bool, error) {
+	if readyEndpoint == "" {
+		readyEndpoint = readinessEndpoint
+	}
```

Comment:
IMHO, it would be clearer to move this to rght before `isEndpointReady()` is called. @caseydavenport WDYT?

Reply:
Added a comment below - I think that conceptually, we should do this the other way:

```go
// If the user specified any readienss gates, run them and ensure they pass. 
for endpoint := range config.ReadinessGates {
  err := pollReadinessEndpoint(endpoint)
  if err != nil {
    return err
  } 
}

Reply:
Fixed